hole/play.go

```diff
@@ -444,9 +444,16 @@ func play(
 
 	stdoutBytes := stdout.Next(maxLength)
 
-	// Postprocess sed output to turn null bytes into newlines.
-	if lang.ID == "sed" {
-		stdoutBytes = bytes.ReplaceAll(stdoutBytes, []byte("\x00"), []byte("\n"))
+	// Postprocess output in apl or sed.
+	// Convert apl's carriage returns or sed's null bytes to newlines.
+	if lang.ID == "apl" || lang.ID == "sed" {
+		stdoutByte := "\r"
+
+		if lang.ID == "sed" {
+			stdoutByte = "\x00"
+		}
+
+		stdoutBytes = bytes.ReplaceAll(stdoutBytes, []byte(stdoutByte), []byte("\n"))
 	}
```

Comment:
Shoutout to @ovs-code for the [recommendation](https://github.com/code-golf/code-golf/pull/1261#discussion_r1725168280) to temporarily do some postprocessing of the output prior to displaying.

This commit combines two separate and nearly identical blocks into one. Besides reducing repetition, this commit implements the aforementioned recommendation.